pkg/metrics/statsd/statsd.go

```diff
@@ -0,0 +1,216 @@
+	} else if err != nil {
+		return nil, err
+	}
+
+	return clientConfig, nil
+}
+
+// Close closes the connection
+func (c *clientConfig) Close() {
+	c.conn.Close()
+}
+
+// Clone clones the connection
+func (c *clientConfig) Clone(opts ...metrics.Option) metrics.Interface {
+	// defaults
+	cConfig := &metrics.ClientConfig{
+		Account:   os.Getenv("METRICS_ACCOUNT"),
+		Namespace: os.Getenv("METRICS_NAMESPACE"),
+		Dims: map[string]string{
+			"location": c.env.Location(),
+		},
+		Prefix: defaultPrefix,
+	}
+	// apply opts input
+	for _, o := range opts {
+		o(cConfig)
+	}
+
+	clone := &clientConfig{
+		ClientConfig: *cConfig,
+	}
+	clone.config = c.config
+	return clone
+}
+
+// EmitFloat records float information
+func (c *clientConfig) EmitFloat(stat string, value float64) error {
+	f := float{
+		Generic: Generic{
+			Account:   c.ClientConfig.Account,
+			Namespace: c.ClientConfig.Namespace,
+			Dims:      c.ClientConfig.Dims,
+			TS:        Now(),
+			Metric:    fmt.Sprintf("%s.%s", c.ClientConfig.Prefix, stat),
+		},
+		Value: float64(value),
+	}
+	b, err := f.Marshal()
+	if err != nil {
+		return err
+	}
+	_, err = c.Send(b)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *clientConfig) EmitFloatWithDims(stat string, value float64, dims ...string) error {
+	opt := metrics.Dimension(dims...)
+	cConfigCopy := c.ClientConfig
+	opt(&cConfigCopy)
+
+	f := float{
+		Generic: Generic{
+			Account:   c.ClientConfig.Account,
+			Namespace: c.ClientConfig.Namespace,
+			Dims:      cConfigCopy.Dims,
+			TS:        Now(),
+			Metric:    fmt.Sprintf("%s.%s", c.ClientConfig.Prefix, stat),
+		},
+		Value: float64(value),
+	}
+	b, err := f.Marshal()
+	if err != nil {
+		return err
+	}
+	_, err = c.Send(b)
+	if err != nil {
+		return err
+	}
+	return nil
+
+}
+
+// EmitGauge records gauge information
+func (c *clientConfig) EmitGauge(stat string, value int64) error {
+	f := gauge{
+		Generic: Generic{
+			Account:   c.ClientConfig.Account,
+			Namespace: c.ClientConfig.Namespace,
+			Dims:      c.ClientConfig.Dims,
+			TS:        Now(),
+			Metric:    fmt.Sprintf("%s.%s", c.ClientConfig.Prefix, stat),
+		},
+		Value: int64(value),
+	}
+	b, err := f.Marshal()
+	if err != nil {
+		return err
+	}
+	_, err = c.Send(b)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *clientConfig) EmitGaugeWithDims(stat string, value int64, dims ...string) error {
+	opt := metrics.Dimension(dims...)
+	cConfigCopy := c.ClientConfig
+	opt(&cConfigCopy)
+
+	f := gauge{
+		Generic: Generic{
+			Account:   c.ClientConfig.Account,
+			Namespace: c.ClientConfig.Namespace,
+			Dims:      cConfigCopy.Dims,
+			TS:        Now(),
+			Metric:    fmt.Sprintf("%s.%s", c.ClientConfig.Prefix, stat),
+		},
+		Value: int64(value),
+	}
+	b, err := f.Marshal()
+	if err != nil {
+		return err
+	}
+	_, err = c.Send(b)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Send data to statsd daemon
+func (c *clientConfig) Send(data []byte) (int, error) {
```

Comment:
calling this Write would be more customary

Reply:
explain, please?